packages/bot/src/transformers/invite.ts

```diff
@@ -1,11 +1,12 @@
-import type { DiscordApplication, DiscordInviteCreate, DiscordInviteMetadata } from '@discordeno/types'
+import type { DiscordApplication, DiscordInviteCreate, DiscordInviteMetadata, DiscordInviteType } from '@discordeno/types'
 import { isInviteWithMetadata, type Application, type Bot, type ScheduledEvent, type User } from '../index.js'
 import type { InviteStageInstance } from './stageInviteInstance.js'
 
 export function transformInvite(bot: Bot, payload: { invite: DiscordInviteCreate | DiscordInviteMetadata; shardId: number }): Invite {
   const props = bot.transformers.desiredProperties.invite
   const invite = {} as Invite
 
+  if (props.type && 'type' in payload.invite) invite.type = payload.invite.type
```

Comment:
Just in case other people wonder like I did: `payload.invite.type` is not used because invite type can be the number 0 which is falsy.

Reply:
That is not even the issue, i would have added !== undefined.

The issue is that DiscordInviteCreate does not have `type`
DiscordInvite does so also DiscordInviteMetadata does, the issue is that i couldn't think of a way to discriminate the union on `DiscordInvite` (with AND without the metadata) from the DiscordInviteCreate